inference-engine/ie_bridges/python/sample/hello_reshape_ssd/hello_reshape_ssd.py

```diff
@@ -0,0 +1,210 @@
+    func = ng.function_from_cnn(net)
+    ops = func.get_ordered_ops()
+    # -----------------------------------------------------------------------------------------------------
+
+    # ------------- 2. Load Plugin for inference engine and extensions library if specified --------------
+    log.info("Device info:")
+    versions = ie.get_versions(args.device)
+    print("{}{}".format(" " * 8, args.device))
+    print("{}MKLDNNPlugin version ......... {}.{}".format(" " * 8, versions[args.device].major,
+                                                          versions[args.device].minor))
+    print("{}Build ........... {}".format(" " * 8, versions[args.device].build_number))
+
+    if args.cpu_extension and "CPU" in args.device:
+        ie.add_extension(args.cpu_extension, "CPU")
+        log.info("CPU extension loaded: {}".format(args.cpu_extension))
+    # -----------------------------------------------------------------------------------------------------
+
+    # --------------------------- 3. Read and preprocess input --------------------------------------------
+
+    print("inputs number: " + str(len(net.input_info.keys())))
+    assert len(net.input_info.keys()) == 1, 'Sample supports clean SSD network with one input and one output'
+    input_name = list(net.input_info.keys())[0]
+
+    for input_key in net.input_info:
+        print("input shape: " + str(net.input_info[input_key].input_data.shape))
+        print("input key: " + input_key)
+        if len(net.input_info[input_key].input_data.layout) == 4:
+            n, c, h, w = net.input_info[input_key].input_data.shape
+            assert n == 1, 'Sample only supports topologies with one input image'
+    
+    h_new, w_new = cv2.imread(args.input).shape[:-1]
+    images = np.ndarray(shape=(n, c, h_new, w_new))
+    images_hw = []
+    for i in range(n):
+        image = cv2.imread(args.input)
+        ih, iw = image.shape[:-1]
+        images_hw.append((ih, iw))
+        log.info("File was added: ")
+        log.info("        {}".format(args.input))
+        image = image.transpose((2, 0, 1))  # Change data layout from HWC to CHW
+        images[i] = image
+    
+    net.reshape({input_name: [n, c, h_new, w_new]})
+    print(net.input_info['data'].input_data.shape)
+
+    # -----------------------------------------------------------------------------------------------------
+
+    # --------------------------- 4. Configure input & output ---------------------------------------------
+    # --------------------------- Prepare input blobs -----------------------------------------------------
+    log.info("Preparing input blobs")
+    assert (len(net.input_info.keys()) == 1 or len(
+        net.input_info.keys()) == 2), "Sample supports topologies only with 1 or 2 inputs"
+    out_blob = next(iter(net.outputs))
+    input_name, input_info_name = "", ""
+
+    for input_key in net.input_info:
+        if len(net.input_info[input_key].layout) == 4:
```

Comment:
Can you please comment on such magic digits at least or use understandable variables or constants.
for example, tensor_dementions = 4

Reply:
Added variable `supported_input_dims` and comment for it